db/db_impl/db_impl_write.cc

```diff
@@ -1933,11 +2098,43 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   }
 
   cfd->mem()->SetNextLogNumber(logfile_number_);
-  cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
+  // If MemPurge activated, purge and delete current memtable.
+  if (immutable_db_options_.experimental_allow_mempurge) {
+    s = MemPurge(cfd, new_mem);
+    if (s.ok()) {
```

Comment:
If MemPurge fails, are you intending to return that error up the chain?  Or should s be a temporary/local variable?

If it is meant to be returned, please add a comment as such -- since there is code below that looks like it is meant to fallback, it looks as if the error is local. 

If the result is not meant to be returned, use a variable other than "s".

Reply:
Good catch - the status was not meant to be returned. I will create another Status variable instead in the next commit.